shared/src/question-generators/math/modTricks.ts

```diff
@@ -0,0 +1,292 @@
+};
+
+export const ModTricks: QuestionGenerator = {
+    id: "modtricks",
+    name: tFunctional(translations, "name"),
+    description: tFunctional(translations, "description"),
+    tags: ["basic math", "modular arithmetic", "modulus", "mod", "arithmetic"],
+    languages: ["en", "de"],
+    author: "Janette Welker",
+    license: "MIT",
+    expectedParameters: [
+        {
+            type: "string",
+            name: "variant",
+            allowedValues: ["simple", "reduction", "inverse", "exponentiation", "crt"],
+        },
+    ],
+
+    generate: (lang, parameters, seed) => {
+        const path = serializeGeneratorCall({
+            generator: ModTricks,
+            lang,
+            parameters,
+            seed,
+        });
+
+        const random = new Random(seed);
+        const variant = parameters.variant as "simple" | "reduction" | "inverse" | "exponentiation" | "crt";
+
+        let question;
+        let testing;
+
+        switch (variant) {
+        case "simple":
+            ({ question, testing } = generateSimpleQuestion(lang, path, random));
+            break;
+        case "reduction":
+            ({ question, testing } = generateReductionQuestion(lang, path, random));
+            break;
+        case "inverse":
+            ({ question, testing } = generateInverseQuestion(lang, path, random));
+            break;
+        case "exponentiation":
+            ({ question, testing } = generateExponentiationQuestion(lang, path, random));
+            break;
+        case "crt":
+            ({ question, testing } = generateCRTQuestion(lang, path, random));
+            break;
+        default:
+            throw new Error("Unknown variant");
+        }
+
+        return {
+        question,
+        testing, // allows for unit tests
+        };
+    },
+};
+
+// Simple
+function generateSimpleQuestion(lang: Language, path: string, random: Random) {
+    const a = random.int(0, 19);
+    const b = random.int(2, 20);
+
+    const question: FreeTextQuestion = {
+        type: "FreeTextQuestion",
+        name: ModTricks.name(lang),
+        path: path,
+        text: t(translations, lang, "simpleQuestion", { a: String(a), b: String(b) }),
+        feedback: getSimpleFeedbackFunction(lang, a, b),
+    };
+
+    const testing = { a, b };
+    return { question, testing };
+}
+
+function getSimpleFeedbackFunction(lang: Language, a: number, b: number): FreeTextFeedbackFunction {
+    return ({ text }) => {
+        // parse to Float to correctly recognize non-integer numbers
+        const userAnswer = parseFloat(text.trim());
+        if (isNaN(userAnswer) || !Number.isInteger(userAnswer)) {
+            return { correct: false, feedbackText: t(translations, lang, "feedbackInvalid") };
+        }
+        return (userAnswer - a) % b === 0
+            ? { correct: true, feedbackText: t(translations, lang, "feedbackCorrect") }
+            : { correct: false, feedbackText: t(translations, lang, "feedbackIncorrect") };
```

Comment:
Maybe add: `correctAnswer: a.toString()` or `...(a * random.int(1, 5).toString())`, such that the user can see a correct solution, but maybe only `a` is to obvious